File: src/Swift.Bindings/tests/IntegrationTests/SmokeTests/SmokeTests.swift
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import Foundation

@frozen
public struct FrozenStruct {
    let x: Int32
    let y: Int32

    public init(x: Int32, y: Int32) {
        self.x = x
        self.y = y
    }

    public func add() -> Int32 {
        return x + y
    }
}

// Add non-frozen struct

File: src/Swift.Bindings/tests/IntegrationTests/SmokeTests/SmokeTests.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

using Xunit;
using Swift.SmokeTests;

namespace BindingsGeneration.Tests
{
    public class SmokeTests : IClassFixture<SmokeTests.TestFixture>
    {
        private readonly TestFixture _fixture;

        public SmokeTests(TestFixture fixture)
        {
            _fixture = fixture;
        }

        public class TestFixture
        {
            static TestFixture()
            {
                InitializeResources();
            }

            private static void InitializeResources()
            {
                // Initialize
            }
        }

        [Fact]
        public static void TestFrozenStruct()
        {
            FrozenStruct f = new FrozenStruct(40, 2);
            Assert.Equal(42, f.add());
        }
        [Fact]
        public static void TestNonFrozenStruct()
        {
            // Add test for non-frozen struct
        }
    }
}
